docs/markdown/Vala.md

```diff
@@ -36,9 +36,21 @@ foo_dep = dependency('foo') # 'foo.vapi' will be resolved in './vapi/foo.vapi'
 executable('app', 'app.vala', dependencies: [glib_dep, gobject_dep, foo_dep])
 ```
 
+In this case, make sure that the VAPI name correspond to the pkg-config file. 
+
+If no pkg-config file is provided, you must use `find_library`. Using`declare_dependency` is cleaner because it does not require passing both dependency objects to the target.
```

Comment:
It's in the custom VAPI section, so I'm sort of assuming that the user will be providing it. I can reword this to cover the case of a VAPI without a corresponding pkg-config file like `posix.vapi`.